packages/components/modals/data-resource/modal-status.tsx

```diff
@@ -0,0 +1,164 @@
+    icon: ExclamationCircleIcon,
+    desc: t(
+      'We could not connect to the {{bucketName}}. You can try again or connect to existing data sources',
+      {
+        bucketName: name,
+      }
+    ),
+    className: 'data-resource-modal__error-icon',
+  },
+});
+
+const STATUS_CONDITIONS = {
+  [NS_PROGRESS.CREATING]: 'warning',
+  [NS_PROGRESS.READY]: 'success',
+  [NS_PROGRESS.REJECTED]: 'danger',
+};
+
+const STATUS_FOOTER = (
+  t: TFunction,
+  onCancel: FuncType,
+  onTryAgain: FuncType
+) => ({
+  [NS_PROGRESS.CREATING]: [
+    {
+      id: 'creating-action-add',
+      label: t('Add'),
+      type: ButtonType.submit,
+      variant: ButtonVariant.primary,
+      disable: true,
+      loading: true,
+    },
+    {
+      id: 'creating-action-cancel',
+      label: t('Cancel'),
+      type: ButtonType.submit,
+      variant: ButtonVariant.secondary,
+      onClick: onCancel,
+    },
+  ],
+  [NS_PROGRESS.READY]: [
+    {
+      id: 'ready-action-finish',
+      label: t('Finish'),
+      type: ButtonType.submit,
+      variant: ButtonVariant.primary,
+      onClick: onCancel,
+    },
+  ],
+  [NS_PROGRESS.REJECTED]: [
+    {
+      id: 'rejected-action-finish',
+      label: t('Finish'),
+      type: ButtonType.submit,
+      variant: ButtonVariant.primary,
+      onClick: onCancel,
+    },
+    {
+      id: 'rejected-action-try-again',
+      label: t('Try again'),
+      type: ButtonType.submit,
+      variant: ButtonVariant.secondary,
+      onClick: onTryAgain,
+    },
+  ],
+});
+
+const DataResourceStatus: React.FC<DataResourceStatusProps> = React.memo(
+  ({ name, status, onCancel, onTryAgain, data, error }) => {
+    const { t } = useTranslation();
+    const statusObj = STATUS_BODY(t, name)[status];
+    const errorMessage =
+      data?.status?.conditions?.[0]?.message || error?.message;
+    return (
+      <>
+        <EmptyState>
+          <EmptyStateIcon
+            icon={statusObj.icon}
+            className={statusObj.className}
+          />
+          <EmptyStateBody data-test="empty-state-body">
+            {statusObj.desc}
+          </EmptyStateBody>
+          {errorMessage && (
+            <Alert
+              variant={STATUS_CONDITIONS[status]}
+              title=""
+              customIcon={<></>}
+              isInline
+            >
+              {errorMessage}
+            </Alert>
+          )}
+        </EmptyState>
+        <ActionGroup className="pf-c-form pf-c-form__actions--left">
+          {STATUS_FOOTER(t, onCancel, onTryAgain)[status].map((buttonProp) => {
+            return (
+              <Button
+                type={buttonProp?.type}
+                variant={buttonProp?.variant}
+                isDisabled={buttonProp?.disable}
+                key={buttonProp?.id}
+                data-test-id={buttonProp?.id}
+                onClick={buttonProp?.onClick}
+                isLoading={buttonProp?.loading}
+              >
+                {buttonProp.label}
```

Comment:
no need ? for label

Reply:
I will add it... it is just label is present in all the objects I have initialised above...
I will use proper types...